ydb/core/tx/schemeshard/ut_backup_collection/ut_backup_collection.cpp

```diff
@@ -2547,4 +2547,127 @@ Y_UNIT_TEST_SUITE(TBackupCollectionTests) {
+            NLs::PathNotExist
+        });
+        TestDescribeResult(DescribePrivatePath(runtime, "/MyRoot/TestTable/" + firstCdcStreamName + "/streamImpl"), {
+            NLs::PathNotExist
+        });
+
+        auto tableDesc2 = DescribePrivatePath(runtime, "/MyRoot/TestTable", true, true);
+        UNIT_ASSERT(tableDesc2.GetPathDescription().HasTable());
+        UNIT_ASSERT_VALUES_EQUAL(tableDesc2.GetPathDescription().GetTable().CdcStreamsSize(), 1);
+
+        TString secondCdcStreamName = tableDesc2.GetPathDescription().GetTable().GetCdcStreams(0).GetName();
+        UNIT_ASSERT_C(secondCdcStreamName.EndsWith("_continuousBackupImpl"), 
+            "New CDC stream should end with '_continuousBackupImpl', got: " + secondCdcStreamName);
+        UNIT_ASSERT_C(firstCdcStreamName != secondCdcStreamName, 
+            "CDC stream name should change after rotation");
+
+        TestDescribeResult(DescribePrivatePath(runtime, "/MyRoot/TestTable/" + secondCdcStreamName), {
+            NLs::PathExist,
+            NLs::StreamMode(NKikimrSchemeOp::ECdcStreamModeUpdate),
+            NLs::StreamFormat(NKikimrSchemeOp::ECdcStreamFormatProto),
+            NLs::StreamState(NKikimrSchemeOp::ECdcStreamStateReady),
+        });
+
+        runtime.AdvanceCurrentTime(TDuration::Seconds(2));
+
+        TestBackupIncrementalBackupCollection(runtime, ++txId, "/MyRoot",
+            R"(Name: ".backups/collections/RotationTestCollection")");
+        env.TestWaitNotification(runtime, txId);
+
+        env.SimulateSleep(runtime, TDuration::Seconds(5));
+
+        TestDescribeResult(DescribePrivatePath(runtime, "/MyRoot/TestTable/" + secondCdcStreamName), {
+            NLs::PathNotExist
+        });
+
+        auto tableDesc3 = DescribePrivatePath(runtime, "/MyRoot/TestTable", true, true);
+        UNIT_ASSERT(tableDesc3.GetPathDescription().HasTable());
+        UNIT_ASSERT_VALUES_EQUAL(tableDesc3.GetPathDescription().GetTable().CdcStreamsSize(), 1);
+
+        TString thirdCdcStreamName = tableDesc3.GetPathDescription().GetTable().GetCdcStreams(0).GetName();
+        UNIT_ASSERT_C(thirdCdcStreamName != secondCdcStreamName, 
+            "CDC stream name should change after second rotation");
+
+        TestBackupBackupCollection(runtime, ++txId, "/MyRoot",
+            R"(Name: ".backups/collections/RotationTestCollection")");
+        env.TestWaitNotification(runtime, txId);
+
+        env.SimulateSleep(runtime, TDuration::Seconds(5));
+
+        // Shouldn't exist, should rotate
+        TestDescribeResult(DescribePrivatePath(runtime, "/MyRoot/TestTable/" + thirdCdcStreamName), {
+            NLs::PathNotExist
+        });
+
+        auto tableDesc4 = DescribePrivatePath(runtime, "/MyRoot/TestTable", true, true);
+        UNIT_ASSERT(tableDesc4.GetPathDescription().HasTable());
+        // test crashed, should be just 1 changefeed
+        UNIT_ASSERT_VALUES_EQUAL(tableDesc4.GetPathDescription().GetTable().CdcStreamsSize(), 1);
```

Comment:
as far as you changing stream creation logic, ensure the data captured correctly for newly created streams (e.g. take another backup and check new data is in backup table)